feat : add api

pooja-ces opened the pull request.

codecrafter-a committed:
feat : add api

vaishali3012 commented on src/providers/staticData.ts:
```diff
@@ -0,0 +1,515 @@
+    schedule: string;
+    time: string;
+    status: number;
+    history: HistoryEntry[];
```
can we make a base interface having time and desc

vaishali3012 commented on src/lib/api.ts:
```diff
@@ -0,0 +1,4 @@
+export async function fetchApiStatus(apiName: string): Promise<number> {
+    const response = await fetch(`/api/status/${apiName}`);
```
add try catch

pooja-ces committed:
fix:-
=> i have make a base interface IHistoryData in staticData file
=> add api in try and catch

Sandeep-Divi commented on src/api/client.tsx:
```diff
@@ -0,0 +1,68 @@
+
```
Since, it is next.js application we can write Next.js apis using route handlers. Please go through the below docs for reference - 
https://nextjs.org/docs/app/building-your-application/routing/route-handlers
https://nextjs.org/docs/pages/building-your-application/routing/api-routes

pooja-ces replied:
fixed

pooja-ces committed:
fix =>
=> I have add  Next.js apis using route handler

pooja-ces committed:
Merge branch 'feature/ui-improvements' into feature/add-api
